localstack-core/localstack/testing/pytest/stepfunctions/utils.py

```diff
@@ -486,6 +532,120 @@ def create_and_record_logs(
     )
 
 
+def launch_and_record_sync_execution(
+    stepfunctions_client,
+    sfn_snapshot,
+    state_machine_arn,
+    execution_input,
+):
+    exec_resp = stepfunctions_client.start_sync_execution(
+        stateMachineArn=state_machine_arn,
+        input=execution_input,
+    )
+    sfn_snapshot.add_transformer(sfn_snapshot.transform.sfn_sm_sync_exec_arn(exec_resp, 0))
+    sfn_snapshot.match("start_execution_sync_response", exec_resp)
+
+
+def create_and_record_express_sync_execution(
+    stepfunctions_client,
+    create_iam_role_for_sfn,
+    create_state_machine,
+    sfn_snapshot,
+    definition,
+    execution_input,
+):
+    snf_role_arn = create_iam_role_for_sfn()
+    sfn_snapshot.add_transformer(RegexTransformer(snf_role_arn, "sfn_role_arn"))
+
+    creation_response = create_state_machine(
+        name=f"express_statemachine_{short_uid()}",
+        definition=definition,
+        roleArn=snf_role_arn,
+        type="EXPRESS",
+    )
+    state_machine_arn = creation_response["stateMachineArn"]
+    sfn_snapshot.add_transformer(sfn_snapshot.transform.sfn_sm_create_arn(creation_response, 0))
+    sfn_snapshot.match("creation_response", creation_response)
+
+    launch_and_record_sync_execution(
+        stepfunctions_client,
+        sfn_snapshot,
+        state_machine_arn,
+        execution_input,
+    )
+
+
+def launch_and_record_express_async_execution(
+    aws_client,
+    sfn_snapshot,
+    state_machine_arn,
+    log_group_name,
+    execution_input,
+):
+    start_execution = aws_client.stepfunctions.start_execution(
+        stateMachineArn=state_machine_arn, input=execution_input
+    )
+    sfn_snapshot.add_transformer(sfn_snapshot.transform.sfn_sm_express_exec_arn(start_execution, 0))
+    execution_arn = start_execution["executionArn"]
+
+    event_list = await_on_execution_logs(
+        aws_client, log_group_name, validation_function=_is_last_history_event_terminal
+    )
+    # Snapshot only the end event, as AWS StepFunctions implements a flaky approach to logging previous events.
```

Comment:
This is very unfortunate indeed. However, since our runtime is still a STANDARD one, which is thoroughly tested, this becomes more of a question of whether we report the logs to the LoggingConfiguration as expected. For STANDARD workflows we actually know which logs to expect, and can handle longer waits or retries until all events are recorded, and later sanitise them. This takes place in all tests that rely on `localstack.testing.pytest.stepfunctions.utils.launch_and_record_logs`, mainly those in `aws.services.stepfunctions.v2.logs.test_logs.TestLogs`. However we cannot easily do this for EXPRESS workflows as obtaining the expected event list would require sampling these form a twin STANDARD workflow.
For EXPRESS workflows it seems like we want to ensure that the state machine is run fully. In this case recording the last event seems enough. It includes information about the output, success, and also a counter of evaluation steps that were executed during runtime (event id).